master/internal/api_checkpoint.go

```diff
@@ -28,6 +37,59 @@ func (a *apiServer) GetCheckpoint(
 	}
 }
 
+func (a *apiServer) DeleteCheckpoints(
+	ctx context.Context,
+	req *apiv1.DeleteCheckpointsRequest) (*apiv1.DeleteCheckpointsResponse, error) {
+	deleteCheckpoints := req.CheckpointUuids
+	dCheckpointsInModelRegistry, _ := a.m.db.GetDeleteCheckpointsInModelRegistry(deleteCheckpoints)
+	// return 400 if model registry checkpoints and include all the model registry checkpoints
+
+	if len(dCheckpointsInModelRegistry) > 0 {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"This subset of list of checkpoints provided are in the model registry: %v cannot be deleted.",
+			dCheckpointsInModelRegistry)
+	}
+
+	jsonVCheckpoints, _ := json.Marshal(deleteCheckpoints)
+	taskSpec := *a.m.taskSpec
+
+	addr := actor.Addr(fmt.Sprintf("delete-checkpoints-list-gc-%s", uuid.New().String()))
+
+	curUser, _, err := grpcutil.GetUser(ctx, a.m.db, &a.m.config.InternalConfig.ExternalSessions)
+	if err != nil {
+		return nil, err
+	}
+
+	jobID := model.NewJobID()
+	if err := a.m.db.AddJob(&model.Job{
+		JobID:   jobID,
+		JobType: model.JobTypeExperiment,
+		OwnerID: &curUser.ID,
+	}); err != nil {
+		return nil, errors.Wrapf(err, "persisting new job")
+	}
+
+	if gcErr := a.m.system.MustActorOf(addr, &checkpointGCTask{
+		taskID:            model.NewTaskID(),
+		jobID:             jobID,
+		jobSubmissionTime: time.Now().UTC().Truncate(time.Millisecond),
+		GCCkptSpec: tasks.GCCkptSpec{
+			Base:         taskSpec,
+			ExperimentID: 0,
+			LegacyConfig: expconf.LegacyConfig{},
```

Comment:
ah.. I think this isn't quite going to work. we're going to need to group the requested checkpoints by experiment id and do a job per group, with the legacy config properly included, else the GC job won't work in most cases.

Reply:
and when this is done, just set experiment id, above.